### PR TITLE
SCSS modules

### PR DESCRIPTION
**Pull request by wongjn**
Dart Sass 1.23.0 dropped recently and with it, [a new module system](http://sass.logdown.com/posts/7858341-the-module-system-is-launched).

This pull request adds parsing for:
- [`@use`](https://sass-lang.com/documentation/at-rules/use)
- [`@forward`](https://sass-lang.com/documentation/at-rules/forward)
- `module.$variables`
- `module.functions()`
- `@include module.includes`

Related to microsoft/vscode#81943.

### CODE REVIEW

**Comment by CarterPape** [comment]
thank you for thiss

File: src/test/scss/parser.test.ts [hunk 1]
```diff
@@ -338,7 +539,7 @@ suite('SCSS - Parser', () => {
 
 	test('Nested Ruleset', function () {
 		let parser = new SCSSParser();
-		assertNode('.class1 { $let: 1; .class { $let: 2; three: $let; let: 3; } one: $let; }', parser, parser._parseRuleset.bind(parser));
+		assertNode('.class1 { $let: 1; .class { $let: 2; three: $let; let: 3; four: m.$foo } one: $let; }', parser, parser._parseRuleset.bind(parser));
```
Comment (octref) [inline src/test/scss/parser.test.ts]:
```suggestion
		assertNode('.class1 { $let: 1; .class { $let: 2; three: $let; let: 3; } one: $let; }', parser, parser._parseRuleset.bind(parser));
		assertNode('.class1 { $let: 1; .class { $let: m.$foo; } one: $let; }', parser, parser._parseRuleset.bind(parser));
```

Think it's better to keep the tests orthogonal.

File: src/test/scss/parser.test.ts [hunk 2]
```diff
@@ -324,7 +525,7 @@ suite('SCSS - Parser', () => {
 
 	test('Ruleset', function () {
 		let parser = new SCSSParser();
-		assertNode('.selector { prop: erty $let 1px; }', parser, parser._parseRuleset.bind(parser));
+		assertNode('.selector { prop: erty $let 1px m.$foo; }', parser, parser._parseRuleset.bind(parser));
```
Comment (octref) [inline src/test/scss/parser.test.ts]: Same as the other orthogonal comment, just add a new one only testing `'.selector { prop: m.$foo; }'

**Comment by wongjn** [comment]
Done.

---

Separately while working on this, I noticed some things that I think are out of scope for this pull request:

**src/test/scss/scssNavigation.test.ts:53**:

```diff
 					if (stats.isFile()) {
 						type = FileType.File;
-					} else if (stats.isDirectory) {
+					} else if (stats.isDirectory()) {
 						type = FileType.Directory;
-					} else if (stats.isSymbolicLink) {
+					} else if (stats.isSymbolicLink()) {
 						type = FileType.SymbolicLink;
 					}
```

Typescript threw some errors here, saying that these were probably meant to be method calls.

**src/services/cssNavigation.ts:14**

```diff
-import { endsWith, startsWith } from '../utils/strings';
+import { startsWith } from '../utils/strings';
```

`endsWith` is unused.

**src/services/scssNavigation.ts:24**

```diff
-	public findDocumentLinks(
-		document: TextDocument,
-		stylesheet: nodes.Stylesheet,
-		documentContext: DocumentContext
-	): DocumentLink[] {
-		return super.findDocumentLinks(document, stylesheet, documentContext);
-	}
```

I don't believe this method override is needed if it is only calling `super`.

**Comment by octref** [comment]
Thanks again, I fixed them up with
https://github.com/microsoft/vscode-css-languageservice/commit/3ebd684a6c1f2e62d5e1502545cabc73e95f0d97

**Pull request merged**